atlasdb-client/src/test/java/com/palantir/atlasdb/schema/cleanup/StreamStoreCleanupMetadataTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.schema.cleanup;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+
+import com.palantir.atlasdb.protos.generated.SchemaMetadataPersistence;
+import com.palantir.atlasdb.table.description.ValueType;
+
+public class StreamStoreCleanupMetadataTest {
+    @Test
+    public void canSerializeAndDeserializePreservingHashedComponentsAndValueTypes() {
```

Comment:
Perhaps this test and `CleanupMetadataTest.canInvokeMethodsOnVisitor` should be parameterised?

Reply:
Hmm, yeah they could be. I'll look into this (`0-2` is a common range for checking things on numbers of hashed components!)

Reply:
The main benefit of doing this would be that you get to see which type(s) and number(s) of hashed components failed, so could be worth doing. There is a little bit more complexity in the test code unfortunately but I think it's still pretty manageable.

Reply:
Done. I didn't bother to do this for the cleanup metadata one as there are only three possible values and the complexity isn't so nice - but can imagine the serialization one could be a horrible mess as it goes over 33 (!) possibilities